packages/x-components/src/components/base-grid.vue

```diff
@@ -114,7 +114,7 @@
        *
        * @internal
        */
-      const injectedListItems = inject<Ref<ListItem[]>>(LIST_ITEMS_KEY as string);
+      const injectedListItems = inject<Ref<ListItem[]> | []>(LIST_ITEMS_KEY as string, []);
```

Comment:
Array type inside `Ref`: `inject<Ref<ListItem[] | []>>`

Reply:
Hi, wasn't sure about it as ID warned: 
![Captura de pantalla 2024-06-05 a les 18 28 07](https://github.com/empathyco/x/assets/21217131/b15a8068-b23e-4e4f-b9ff-dbe83734ae4d)

Reply:
☝️ Set ref([]) as default value made the job, but , what do we want to do with the console.warn? Keep it or shall I change tests?? 🙏 @joseacabaneros

Reply:
I'm checking how we implement this same case in other injections, and it is common to use `undefined` as default value for the `inject`. 
With the aim of align the code base, could you change it by `undefined`? 
It is all about alignment, yours is also perfectly valid  ✌🏾  

```TS
const injectedListItems = inject<Ref<ListItem[]> | undefined>(
  LIST_ITEMS_KEY as string,
  undefined
);
```